src/overtone/sc/sclang.clj

```diff
@@ -0,0 +1,416 @@
+         app-clock-end-coll ["});"]]
+     #_(println code-str)
+     (format (->> (concat
+                   (when boot boot-init-coll)
+                   app-clock-init-coll
+                   [""
+                    "try {this.interpret("
+                    "%s"
+                    ") }"
+                    "  { |error|"
+                    "    \""
+                    ""
+                    "ERROR"
+                    "----------------------------------"
+                    "\".postln;"
+                    "    error.reportError;"
+                    "    \"-------------------------------------"
+                    ""
+                    "\".postln;"
+                    "  };"
+                    ""]
+                   app-clock-end-coll
+                   (when boot boot-end-coll))
+                  (str/join "\n"))
+             (pr-str code-str)))))
+
+(defn exec!
+  "Execute a sclang script in the background.
+
+  It returns a `babashka.process/process`, `out` and `err` are redirected
+  to stdout.
+
+  Call `stop-procs!` or `babaskha.process/destroy-tree` to kill the
+  returned process.
+
+  E.g.
+
+    (exec! [:. :SynthDef :help])"
+  [sc-clj]
+  (stop-procs!)
+  (let [temp-scd (io/file (str ".overtone/sc/" "_" (random-uuid) ".scd"))
+        _port (or (:port @ov.conn/connection-info*)
+                  (:port (:opts @ov.conn/connection-info*)))
+        str (-wrap-code-with-interpreter (transpile sc-clj))]
+    (io/make-parents temp-scd)
+    (spit temp-scd str)
+    (let [proc (proc/process {:out *out* :err *err*} (sclang-path) temp-scd)]
+      (swap! *procs conj proc)
+      proc)))
+#_(exec! [:. :SynthDef :help])
+;; If you don't have `FoaRotate` installed, you should see an error in the REPL output.
+#_(exec! [:. :FoaRotate :help])
+
+(defn help
+  "Open SC help window for a object."
+  [obj-k]
+  (exec! [:. obj-k :help]))
+#_(help :SynthDef)
+
+(defn- check-proc!
+  [proc args]
+  (loop [counter 5]
+    (cond
+      (zero? counter)
+      (throw (ex-info (str "Process had an error, check the stdout, also check "
+                           "that you have SuperCollider installed, https://supercollider.github.io/downloads")
+                      (merge args {:proc (proc/destroy-tree proc)})))
+
+      (proc/alive? proc)
+      (do (Thread/sleep 200)
+          (recur (dec counter)))
+
+      :else
+      @proc)))
+
+(defn -synthdef-save!
+  "This is an internal function, read the comments in the code to understand
+  what it's doing."
+  [{:keys [synthdef-name file-path resource-path
+           metadata-file-path metadata-resource-path
+           synthdef-str sc-clj]
+    :as args}]
+  (if (and (io/resource resource-path)
+           (io/resource metadata-resource-path)
+           (= (hash sc-clj) (-> (io/resource metadata-resource-path)
+                                slurp
+                                edn/read-string
+                                :code-hash)))
```

Comment:
There's a lot of corner cases with invalidating edn caches with hashes. It might be best to start with just comparing edn with `=` (or look into the many library that tackle this issue).

The only thing to keep in mind there is the `spit` call should look like this:

```clojure
(spit metadata-file-path (binding [*print-length* nil, *print-level* nil]
                           (pr-str {:edn sc-clj})))
```

Reply:
Done, thanks \o I've thought about using just `=`, and indeed it's safer \o